mobile/lib/shared/ui/immich_image.dart

```diff
@@ -1,143 +1,178 @@
+        isThumbnail: isThumbnail,
+      );
+    }
+
+    if (useLocal(asset) && isThumbnail) {
+      return AssetEntityImageProvider(
+        asset.local!,
+        isOriginal: false,
+        thumbnailSize: ThumbnailSize.square(thumbnailSize),
+      );
+    } else if (useLocal(asset) && !isThumbnail) {
+      return ImmichLocalImageProvider(
+        asset: asset,
+      );
+    } else {
+      return ImmichRemoteImageProvider(
+        assetId: asset.remoteId!,
+        isThumbnail: isThumbnail,
+      );
+    }
+  }
+
+  static bool useLocal(Asset asset) =>
+      !asset.isRemote ||
+      asset.isLocal && !Store.get(StoreKey.preferRemoteImage, false);
+
+  @override
+  createState() => _ImmichImageState();
+}
+
+/// Renders an Asset using local data if available, else remote data
+class _ImmichImageState extends State<ImmichImage> {
+  // Creating the Uint8List from the List<bytes> during each build results in flickers during
+  // the fade transition. Calculate the hash in the initState and cache it for further builds
+  Uint8List? thumbHashBytes;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.isThumbnail && widget.asset?.thumbhash != null) {
+      final bytes = widget.asset!.thumbhash! as Uint8List;
+      final rgbaImage = thumbhash.thumbHashToRGBA(bytes);
+      thumbHashBytes = thumbhash.rgbaToBmp(rgbaImage);
+    }
+  }
 
   @override
   Widget build(BuildContext context) {
-    if (this.asset == null) {
+    if (widget.asset == null) {
       return Container(
         decoration: const BoxDecoration(
           color: Colors.grey,
         ),
         child: SizedBox(
-          width: width,
-          height: height,
+          width: widget.width,
+          height: widget.height,
           child: const Center(
             child: Icon(Icons.no_photography),
           ),
         ),
       );
     }
-    final Asset asset = this.asset!;
-    if (useLocal(asset)) {
-      return Image(
-        image: localImageProvider(asset, size: preferredLocalAssetSize),
-        width: width,
-        height: height,
-        fit: fit,
-        frameBuilder: (context, child, frame, wasSynchronouslyLoaded) {
-          if (wasSynchronouslyLoaded || frame != null) {
-            return child;
-          }
 
-          // Show loading if desired
-          return Stack(
-            children: [
-              if (useGrayBoxPlaceholder)
-                const SizedBox.square(
-                  dimension: 250,
-                  child: DecoratedBox(
-                    decoration: BoxDecoration(color: Colors.grey),
-                  ),
-                ),
-              if (useProgressIndicator)
-                const Center(
-                  child: CircularProgressIndicator(),
-                ),
-            ],
+    final Asset asset = widget.asset!;
+
+    return OctoImage(
+      fadeInDuration: const Duration(milliseconds: 200),
+      fadeOutDuration: const Duration(milliseconds: 200),
```

Comment:
This is more of a personal preference, but the following duration seems to create a more seamless fade out animation from the thumb-hash to the actual thumbnail. We don't want the image to fade-in, we just want the placeholder to fade-out instead

```suggestion
      fadeInDuration: const Duration(milliseconds: 0),
      fadeOutDuration: const Duration(milliseconds: 400)
```

Reply:
I like it. Let me try it locally and see how it works.